includes/AMP/Sanitization.php

```diff
@@ -0,0 +1,459 @@
+		/**
+		 * Script element.
+		 *
+		 * @var DOMElement $ordered_script
+		 */
+		foreach ( $ordered_scripts as $ordered_script ) {
+			if ( $previous_node && $previous_node->nextSibling ) {
+				$document->head->insertBefore( $ordered_script, $previous_node->nextSibling );
+				$previous_node = $ordered_script;
+			}
+		}
+	}
+
+	/**
+	 * Returns AMP extension URLs, keyed by extension name.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @return array List of extensions and their URLs.
+	 */
+	protected function get_extension_sources() {
+		$specs = [];
+		// Register all AMP components as defined in the spec.
+		foreach ( AMP_Allowed_Tags_Generated::get_extension_specs() as $extension_name => $extension_spec ) {
+			$src = sprintf(
+				'https://cdn.ampproject.org/v0/%s-%s.js',
+				$extension_name,
+				end( $extension_spec['version'] )
+			);
+
+			$specs[ $extension_name ] = $src;
+		}
+
+		return $specs;
+	}
+
+	/**
+	 * Determine whether AMP dev mode is enabled.
+	 *
+	 * When enabled, the <html> element will get the data-ampdevmode attribute and the plugin will add the same attribute
+	 * to elements associated with the admin bar and other elements that are provided by the `amp_dev_mode_element_xpaths`
+	 * filter.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @return bool Whether AMP dev mode is enabled.
+	 */
+	protected function is_amp_dev_mode() {
+		/**
+		 * Filters whether AMP dev mode is enabled.
+		 *
+		 * When enabled, the data-ampdevmode attribute will be added to the document element and it will allow the
+		 * attributes to be added to the admin bar. It will also add the attribute to all elements which match the
+		 * queries for the expressions returned by the 'web_stories_amp_dev_mode_element_xpaths' filter.
+		 *
+		 * @since 1.1.0
+		 *
+		 * @param bool Whether AMP dev mode is enabled.
+		 */
+		return apply_filters(
+			'web_stories_amp_dev_mode_enabled',
+			(
+				// For the few sites that forcibly show the admin bar even when the user is logged out, only enable dev
+				// mode if the user is actually logged in. This prevents the dev mode from being served to crawlers
+				// when they index the AMP version.
+				( is_admin_bar_showing() && is_user_logged_in() )
+			)
+		);
+	}
+
+	/**
+	 * Returns a list of sanitizers to use.
+	 *
+	 * This is replica of amp_get_content_sanitizers() to avoid
+	 * loading amp-helper-functions.php due to side-effects like
+	 * accessing options from the database, requiring AMP__VERSION,
+	 * and causing conflicts with our own amp_is_request() compat shim.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @return array Sanitizers.
+	 */
+	protected function get_sanitizers() {
+		$sanitizers = [
+			AMP_Script_Sanitizer::class            => [],
+			AMP_Style_Sanitizer::class             => [
+
+				/*
+				 * @todo Enable by default and allow filtering once AMP_Style_Sanitizer does not call AMP_Options_Manager
+				 *       which in turn requires AMP__VERSION to be defined.
```

Comment:
I wonder if php-scoper should replace `AMP__VERSION` with the current version string?

Reply:
Interesting thought. I don‘t think the scoper knows the current version number though?

Reply:
Right. You'd have to supply the version number when you upgrade the dependency I guess. Or you could look up the version in a patcher callback, maybe: https://github.com/humbug/php-scoper#patchers

Reply:
Looking at `AMP_Options_Manager`, it seems like patching `AMP__VERSION` would only help so much. It would still call `get_option` and thus causing a DB query.